code/datums/components/crafting/recipes.dm

```diff
@@ -45,6 +45,14 @@
 /datum/crafting_recipe/proc/on_craft_completion(mob/user, atom/result)
 	return
 
+///Check if the pipe used for atmospheric device crafting is the proper one
+/datum/crafting_recipe/proc/atmos_pipe_check(mob/user, list/collected_requirements)
```

Comment:
a good idea would have been to make a /pipe/ subtype and move all pipes and this proc there like food stuff does

Reply:
long time no see

Reply:
(youre right though)